### PR TITLE
Add tests for the `services/govmomi/extra` package

### LINKED ISSUE #1471: Implement unit test cases for extra package
/area testing

Implement unit tests for the public methods under the `pkg/services/govmomi/extra` package. This package is mainly used to set the cloud-init data which gets leveraged on VM startup.

/help
/good-first-issue

### PR DESCRIPTION
**Pull request by ditsuke**
**What this PR does / why we need it**
This PR adds test cases for the public methods under the services/govmomi/extra package. These methods are used to
set cloud init data (used on VM startup) and to set other configuration keys. The new tests cover:
- [x] `config.SetCustomVMXKeys()`
	Passed a string => string map, this method should append all these key, value pairs to the config as `types.OptionValue` structs.
- [x] `config.SetCloudInitUserData()`, `config.SetCloudInitMetadata()`
	Passed byte arrays, these methods are expected to:
	- Store the data in a `guestinfo.userdata`/`guestinfo.metadata` key with a single pass of base64 encoding.
	- Add a key to indicate the encoding format.
	- Ensure the data passed is encoded with exactly a single pass of base64, which includes checking for existing
	  base64 encoding and undoing them before encoding again with 1x base64.

**Which issue(s) this PR fixes**
Fixes #1471

Signed-off by: ditsuke <ditsuke@protonmail.com>

### CODE REVIEW

File: pkg/services/govmomi/extra/extra_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		It("must set a key to indicate base64 encoding of the data", func() {
+			Expect(*config).To(ContainElement(&types.OptionValue{
+				Key:   encodingKey,
+				Value: "base64",
+			}))
+		})
+	})
+
+	Context(fmt.Sprintf("We call %q with some pre-encoded data (single pass)", methodName), func() {
+		config := getMockConfig()
```
Comment (geetikabatra) [inline pkg/services/govmomi/extra/extra_suite_test.go:120]: Same as above.

File: pkg/services/govmomi/extra/extra_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*config).To(ContainElement(&types.OptionValue{
+				Key:   dataKey,
+				Value: preEncodedData,
+			}))
+		})
+	})
+
+	Context(fmt.Sprintf("We call %q with some pre-encoded data (multiple passes)", methodName), func() {
+		config := getMockConfig()
```
Comment (geetikabatra) [inline pkg/services/govmomi/extra/extra_suite_test.go:134]: Same as above.

**Review by geetikabatra — CHANGES_REQUESTED** [review]

File: pkg/services/govmomi/extra/extra_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	ConfigInitFnTester(func(config *Config, s string) error {
+		return config.SetCloudInitMetadata([]byte(s))
+	},
+		"SetCloudInitMetadata",
+		"guestinfo.metadata",
+		"guestinfo.metadata.encoding",
+	)
+})
+
+func getMockConfig() *Config {
```
[the remark this answers is not in the document]
Comment (ditsuke) [reply pkg/services/govmomi/extra/extra_suite_test.go]: That looks a lot better! Done with 127ef34

File: pkg/services/govmomi/extra/extra_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+var _ = Describe("Config_SetCustomVMXKeys", func() {
+	Context("we try to set custom keys in the config", func() {
+		config := getMockConfig()
+		customConfigKeys := map[string]string{
+			"customKey1": "customVal1",
+			"customKey2": "customKey2",
+		}
+
+		var configKeys []interface{}
+		for k, v := range customConfigKeys {
```
[the remark this answers is not in the document]
Comment (ditsuke) [reply pkg/services/govmomi/extra/extra_suite_test.go]: Thanks! This felt smelly even when I first wrote it:)

File: pkg/services/govmomi/extra/extra_suite_test.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+				Key:   k,
+				Value: v,
+			})
+		}
+
+		It("adds the keys to the config", func() {
+			err := config.SetCustomVMXKeys(customConfigKeys)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*config).To(ConsistOf(configKeys...))
```
[the remark this answers is not in the document]
Comment (ditsuke) [reply pkg/services/govmomi/extra/extra_suite_test.go]: Couldn't agree more! Done with 127ef34

**Pull request merged**